socialite/server/graphql/resolvers/posts.js

```diff
@@ -15,6 +15,74 @@ module.exports = {
                 throw new Error(err);
             }
         },
+        async getTopPosts(){
+            try{
+                const posts = await Post.find().sort({ upvotes: -1 });
+                posts.sort((a, b) => (a.upvotes.length < b.upvotes.length) ? 1 : -1);
+                return posts.slice(0, 5);
+            } catch(err){
+                throw new Error(err);
+            }
+        },
+        async getTopRated(){
+            try{
+                const users = await User.find().sort({ rating: -1 });
+                users.sort((a, b) => (a.rating < b.rating) ? 1 : -1);
+                return users.slice(0, 10);
+            } catch(err){
+                throw new Error(err);
+            }
+        },
+        async getTopAnswered(){
+            try{
+                const users = await User.find().sort({ times_answered: -1 });
+                users.sort((a, b) => (a.times_answered < b.times_answered) ? 1 : -1);
+                return users.slice(0, 10);
+            } catch(err){
+                throw new Error(err);
+            }
+        },
+        async getTopTags(){
+            try{
+                var pretags = [
+                    {'name' : 'Data Structures', 'value' : 0},
+                    {'name' : 'C-programming', 'value' : 0},
+                    {'name' : 'Probability and Statistics', 'value' : 0},
+                    {'name' : 'Database Security', 'value' : 0},
+                    {'name' : 'Science', 'value' : 0},
+                    {'name' : 'Discrete Structures', 'value' : 0},
+                    {'name' : 'Speech Processing', 'value' : 0},
+                    {'name' : 'Computer Vision', 'value' : 0},
+                    {'name' : 'Internships', 'value' : 0},
+                    {'name' : 'Resume', 'value' : 0},
+                    {'name' : 'Computational Linguistics', 'value' : 0},
+                    {'name' : 'Algorithm Analysis & Design', 'value': 0},
+                    {'name' : 'NLP', 'value' : 0}
+                ]
```

Comment:
If we add more tags while scaling, we will have to add tags here too, right? Isn't there a way to fetch all tags and build this array automatically?

Reply:
There might be. However, this is a first implementation and also has a close to n^3 running time. We can optimise this later.